tools/integrations/manage.sh

```diff
@@ -343,22 +332,47 @@ cmd_publish_external() {
   local image_version=$1; shift || error "Missing target (image version) $USAGE"
 
   echo "image $image_name:$image_version"
-
   echo "Publishing and writing to spec cache."
+  echo "Using environment gcloud"
+
+  publish_spec_files "$image_name" "$image_version"
+}
+
+generate_spec_file() {
+  local image_name=$1; shift || error "Missing target (image name)"
+  local image_version=$1; shift || error "Missing target (image version)"
+  local tmp_spec_file=$1; shift || error "Missing target (temp spec file name)"
+  local deployment_mode=$1; shift || error "Missing target (deployment mode)"
+
+  docker run --env DEPLOYMENT_MODE="$deployment_mode" --rm "$image_name:$image_version" spec | \
+      # 1. filter out any lines that are not valid json.
+      jq -R "fromjson? | ." | \
+      # 2. grab any json that has a spec in it.
+      # 3. if there are more than one, take the first one.
+      # 4. if there are none, throw an error.
+      jq -s "map(select(.spec != null)) | map(.spec) | first | if . != null then . else error(\"no spec found\") end" \
+      > "$tmp_spec_file"
+}
+
+publish_spec_files() {
+  local image_name=$1; shift || error "Missing target (image name)"
+  local image_version=$1; shift || error "Missing target (image version)"
+
   # publish spec to cache. do so, by running get spec locally and then pushing it to gcs.
-  local tmp_spec_file; tmp_spec_file=$(mktemp)
-  docker run --rm "$image_name:$image_version" spec | \
-    # 1. filter out any lines that are not valid json.
-    jq -R "fromjson? | ." | \
-    # 2. grab any json that has a spec in it.
-    # 3. if there are more than one, take the first one.
-    # 4. if there are none, throw an error.
-    jq -s "map(select(.spec != null)) | map(.spec) | first | if . != null then . else error(\"no spec found\") end" \
-    > "$tmp_spec_file"
+  local tmp_default_spec_file; tmp_default_spec_file=$(mktemp)
+  local tmp_cloud_spec_file; tmp_cloud_spec_file=$(mktemp)
 
-  echo "Using environment gcloud"
+  # generate oss and cloud spec files
+  generate_spec_file "$image_name" "$image_version" "$tmp_default_spec_file" "OSS"
+  generate_spec_file "$image_name" "$image_version" "$tmp_cloud_spec_file" "CLOUD"
```

Comment:
Ah! so we still need to build 2 spec files.

Not for this hack day project, but maybe the "scalable" thing to do is modify how the SPEC output works, and tag certain params as only being available in certain environments.  This would be helpful too in saying that only cloud has oAuth, for example.

Reply:
I have created this issue to track this: https://github.com/airbytehq/airbyte/issues/13782